client/landing/stepper/declarative-flow/internals/steps-repository/launchpad/step-content.tsx

```diff
@@ -9,11 +13,47 @@ type StepContentProps = {
 	goToStep?: NavigationControls[ 'goToStep' ];
 };
 
-const StepContent = ( { siteSlug, submit, goNext, goToStep }: StepContentProps ) => (
-	<div className="launchpad__content">
-		<Sidebar siteSlug={ siteSlug } submit={ submit } goNext={ goNext } goToStep={ goToStep } />
-		<LaunchpadSitePreview siteSlug={ siteSlug } />
-	</div>
-);
+function sortByRegistrationDate( domainObjectA: ResponseDomain, domainObjectB: ResponseDomain ) {
+	const registrationDateA = domainObjectA.registrationDate;
+	const registrationDateB = domainObjectB.registrationDate;
+
+	if ( registrationDateA > registrationDateB ) {
+		return -1;
+	} else if ( registrationDateA < registrationDateB ) {
+		return 1;
+	}
+	return 1;
+}
+
+const StepContent = ( { submit, goNext, goToStep }: StepContentProps ) => {
+	const site = useSite();
+
+	const { data: allDomains = [] } = useGetDomainsQuery( site?.ID ?? null, {
+		retry: false,
+	} );
+
+	const domains = allDomains.map( createSiteDomainObject );
+
+	// Filter custom domains and sort by registration date.
+	// We want the first domain object to be the most recently registered domain
+	const nonWpcomDomains = domains
+		.filter( ( domain ) => ! domain.isWPCOMDomain )
+		.sort( sortByRegistrationDate );
+
+	const wpcomDomains = domains.filter( ( domain ) => domain.isWPCOMDomain );
+
+	const sidebarURL = nonWpcomDomains?.length
+		? nonWpcomDomains[ 0 ]?.domain
+		: wpcomDomains[ 0 ]?.domain;
+
+	const iFrameURL = wpcomDomains.length ? wpcomDomains[ 0 ]?.domain : nonWpcomDomains[ 0 ]?.domain;
+
+	return (
+		<div className="launchpad__content">
+			<Sidebar siteSlug={ sidebarURL } submit={ submit } goNext={ goNext } goToStep={ goToStep } />
```

Comment:
Using `sidebarURL` for the `siteSlug` prop of the entire sidebar may have some side-effects, such as what @edanzer noted in slack. p1665514707670979/1665513799.968999-slack-CHN6J22MP

It seems the Sidebar component still uses this siteSlug for other things than just the domain previewed to the user above the checklist, such as the URL for writing new posts.

Reply:
I will revert the `siteSlug` prop changes and add an additional prop for the sidebar URL.